src/client/Microsoft.Identity.Client/Platforms/Features/OpenTelemetry/OtelInstrumentation.cs

```diff
@@ -95,13 +95,20 @@ public OtelInstrumentation()
             AuthenticationResultMetadata authResultMetadata,
             ILoggerAdapter logger)
         {
+            var tokenSource = authResultMetadata.TokenSource;
+
+            //if (authResultMetadata.CacheRefreshReason == CacheRefreshReason.ProactivelyRefreshed)
+            //{
+            //    tokenSource = TokenSource.IdentityProvider;
```

Comment:
This dead code seems like a nicer way to infer the token source. Can it be used instead of updateing every background request logic?

Reply:
I added this one first and then used the other way. Thinking again I think this will be a better way since we only need it as IDP for otel. The only reason I used the current approach is because this way we will send token source as cache in `AuthenticationResultMetadata` but pass it as IDP to otel. The current approach will have it consistent.